widget/list.go

```diff
@@ -0,0 +1,405 @@
+	li.ExtendBaseWidget(li)
+
+	li.statusIndicator = canvas.NewRectangle(theme.BackgroundColor())
+
+	objects := []fyne.CanvasObject{li.statusIndicator, li.child}
+
+	return &listItemRenderer{widget.NewShadowingRenderer(objects, widget.ButtonLevel), li, layout.NewHBoxLayout()}
+}
+
+// MinSize returns the size that this widget should not shrink below
+func (li *listItem) MinSize() fyne.Size {
+	li.ExtendBaseWidget(li)
+	return li.BaseWidget.MinSize()
+}
+
+// MouseIn is called when a desktop pointer enters the widget
+func (li *listItem) MouseIn(*desktop.MouseEvent) {
+	li.hovered = true
+	li.Refresh()
+}
+
+// MouseMoved is called when a desktop pointer hovers over the widget
+func (li *listItem) MouseMoved(*desktop.MouseEvent) {
+}
+
+// MouseOut is called when a desktop pointer exits the widget
+func (li *listItem) MouseOut() {
+	li.hovered = false
+	li.Refresh()
+}
+
+// Tapped is called when a pointer tapped event is captured and triggers any tap handler
+func (li *listItem) Tapped(*fyne.PointEvent) {
+	if li.onTapped != nil && !li.Disabled() {
+		li.selected = true
+		li.Refresh()
+		li.onTapped()
+	}
+}
+
+type listItemRenderer struct {
+	*widget.ShadowingRenderer
+
+	item   *listItem
+	layout fyne.Layout
+}
+
+// MinSize calculates the minimum size of a listItem.
+// This is based on the size of the status indicator and the size of the child object.
+func (li *listItemRenderer) MinSize() (size fyne.Size) {
+	li.item.statusIndicator.SetMinSize(fyne.NewSize(theme.Padding(), li.item.child.Size().Height))
+	size = fyne.NewSize(theme.Padding()+li.item.child.Size().Width, li.item.child.Size().Height)
+	return
+}
+
+// Layout the components of the button widget
+func (li *listItemRenderer) Layout(size fyne.Size) {
+	li.LayoutShadow(size, fyne.NewPos(0, 0))
+	li.layout.Layout([]fyne.CanvasObject{li.item.statusIndicator, li.item.child}, size)
+}
+
+func (li *listItemRenderer) Refresh() {
+	if li.item.selected {
+		li.item.statusIndicator.FillColor = theme.FocusColor()
+	} else if li.item.hovered {
+		li.item.statusIndicator.FillColor = theme.HoverColor()
+	} else {
+		li.item.statusIndicator.FillColor = theme.BackgroundColor()
+	}
+	canvas.Refresh(li.item.super())
+}
+
+type listLayout struct {
+	list      *List
+	itemCount int
+}
+
+var _ fyne.Layout = (*listLayout)(nil)
+
+func newListLayout(list *List) fyne.Layout {
+	return &listLayout{list: list}
+}
+
+func (l *listLayout) Layout(objects []fyne.CanvasObject, size fyne.Size) {
+	if l.list.offsetY != 0 {
+		return
+	}
+	y := 0
+	for _, child := range objects {
+		child.Move(fyne.NewPos(0, y))
+		y += l.list.itemMin.Height
+		child.Resize(fyne.NewSize(l.list.size.Width, l.list.itemMin.Height))
+	}
+}
+
+func (l *listLayout) MinSize(objects []fyne.CanvasObject) fyne.Size {
+
+	return fyne.NewSize(l.list.itemMin.Width+theme.Padding(),
+		l.list.itemMin.Height*l.list.Length())
+}
+
+func (l *listLayout) itemAppended(objects []fyne.CanvasObject) {
```

Comment:
It's kindof subtle here but with this being called from appendItem wouldn't appendedItem be better? (and the prepended below)

Reply:
Agreed. Resolved.